tests/core/test_device_simulation.py

```diff
@@ -0,0 +1,130 @@
+from typing import Hashable, Iterable, Mapping, cast
+
+import pytest
+from immutables import Map
+from mock import Mock, create_autospec, patch
+
+from tickit.core.adapter import AdapterConfig, ListeningAdapter
+from tickit.core.components.device_simulation import DeviceSimulation
+from tickit.core.state_interfaces.internal import (
+    InternalStateConsumer,
+    InternalStateProducer,
+    InternalStateServer,
+)
```

Comment:
That doesn't actually link to a comment :confused: 
I'm going to assume you mean that these should be mocks :slightly_smiling_face:

Reply:
Ah yeah, comments can't be linked to until the review is pushed so you need to retroactively link, which I forgot to do. Your assumption would be correct

Reply:
So since I have to pass these mocks as arguments to one of the methods, I could avoid importing these classes by using spec-less mocks. The other alternative is to use the base classes instead? I've just implemented the latter locally and it's a simple substitution.

Reply:
Either mocks which are specked from the Protocol or the Protocol itself is fine